ledger/tracker.go

```diff
@@ -571,10 +576,17 @@ func (tr *trackerRegistry) commitRound(dcc *deferredCommitContext) error {
 	ledgerCommitroundMicros.AddMicrosecondsSince(start, nil)
 
 	if err != nil {
+
 		for _, lt := range tr.trackers {
-			lt.handleUnorderedCommitOrError(dcc)
+			lt.handleCommitError(dcc)
 		}
 		tr.log.Warnf("unable to advance tracker db snapshot (%d-%d): %v", dbRound, dbRound+basics.Round(offset), err)
+
+		// if the error is an IO error, shut down the node.
+		if errors.Is(err, trackerdb.ErrIoErr) {
+			tr.log.Fatalf("Fatal IO error during CommitRound, exiting: %v", err)
```

Comment:
please write a test to trigger this case:
1. Set logger's ExitFunc
2. Add a new tracker: its CommitRound calls a new `TestingMethod` from SQL implementation that has maybeIoError wrapping sqlite's `sqlite3.ErrIoErr`.
3. Double check the SQLDriver returns sqlite3.ErrIoErr as pointer and not value - i.e. double check and write a test for maybeIOErr - other code checks for sqlite error on a value, not on a pointer, so I'm not sure this check actually works, so maybe check `Code` instead?
```
	var sqliteErr sqlite3.Error
	if errors.As(obj, &sqliteErr) {
		return sqliteErr.Code == sqlite3.ErrLocked || sqliteErr.Code == sqlite3.ErrBusy
	}
```

Reply:
@algorandskiy you are right, reading the source of sqlite3 I am seeing that the returned errors use an internal `Code ErrNo`, and *that* is the ErrIOErr value. I am writing a test for this now, and one at the tracker level.

@cce sounds like you talked yourself out of not shutting down on IO error? LMK if not and we can chat about it.

Reply:
* Fixed ErrIOError casting, 
* Added MaybeIOErr testing and included some comments w/ links to show how SQL provides its errors
* Added comment to Exit Handling unit test to highlight why it exists
* Added a tracker test which uses an IOErrTracker to force a sqlite3 IO Error, and confirms that fatal logging (and underlying handler functions) get called on commit